lib/controller.ex

```diff
@@ -56,87 +55,67 @@ defmodule Controller do
     |> fetch_content()
   end
 
-  def fetch_content(%{input: input, view: view} = context) do
-    case view do
-      :welcome ->
-        case input do
-          "I" ->
-            recipe_list = Messages.get_recipe(:all) |> Formatter.numbered_list()
-            %{context | content: recipe_list, view: :index}
-
-          "Q" ->
-            nil
-
-          :welcome ->
-            context
+  def fetch_content(%{input: "Q", view: _}) do
+    nil
+  end
 
-          _ ->
-            error = Messages.get_prompt(:unknown)
-            %{context | error: error}
-        end
+  def fetch_content(%{input: "I", view: _} = context) do
+    content = Messages.get_recipe(:all) |> Formatter.numbered_list()
+    %{context | content: content, view: :index}
+  end
 
-      :index ->
-        case input do
-          "Q" ->
-            nil
-
-          _ ->
-            if Messages.get_recipe(input) do
-              recipe =
-                Messages.get_recipe(input)
-                |> RecipeParser.read_file()
-
-              %{context | content: recipe, view: :view_recipe}
-            else
-              error = Messages.get_recipe(:not_found)
-              %{context | error: error}
-            end
-        end
+  def fetch_content(%{input: _, view: :grocery_list} = context) do
+    fetch_unknown_error(context)
+  end
 
-      :view_recipe ->
-        case input do
-          "G" ->
-            if Messages.get_recipe("1") do
-              grocery_list =
-                Messages.get_recipe("1")
-                |> RecipeParser.parse_grocery_list()
-                |> Formatter.bulleted_list()
-
-              %{context | content: grocery_list, view: :grocery_list}
-            else
-              error = Messages.get_recipe(:not_found)
-              %{context | error: error}
-            end
-
-          "I" ->
-            content = Messages.get_recipe(:all) |> Formatter.numbered_list()
-            %{context | content: content, view: :index}
-
-          "Q" ->
-            nil
-
-          _ ->
-            error = Messages.get_prompt(:unknown)
-            %{context | error: error}
-        end
+  def fetch_content(%{input: input, view: :welcome} = context) do
+    case input do
+      :welcome ->
+        context
 
-      :grocery_list ->
-        case input do
-          "I" ->
-            content = Messages.get_recipe(:all) |> Formatter.numbered_list()
-            %{context | content: content, view: :index}
+      _ ->
```

Comment:
This could be handled like line 67 `def fetch_content(%{input: _, view: :grocery_list} = context) do
`
Where you could do a   `def fetch_content(%{input: _, view: :welcome} = context) do`

Reply:
There turned out to be multiple views that were calling fetch_unknown_error in a case statement with only two cases, so I made it the last fetch_content/1 and took out those case statements.